### ISSUE TITLE
Twitter importer: Media 404 crashes the importer

### ISSUE THREAD

**Issue by remram44** [camlistore/camlistore#866]
If an old tweet only has broken media links, an error is shown and there is no way to get the importer to go further: `error: All media URLs 404s for tweet https://twitter.com/remram44/status/73066309369348096`

**Comment by mpl** [comment]
@bradfitz Is there any reason we don't let the importer run go on when an importTweet error is due to its media being a 404?

**Comment by bradfitz** [comment]
It should keep going as far as it can. The tricky part, though, is retries. If the server had a temporary error, we don't want to ignore it forever and then mark the whole import as successful and never try again in an incremental run later.

On the other hand, we don't want to retry forever either.

What we should probably do is track permanodes which are incomplete and are missing resources, and then try them again in the future on incremental backups, with some sort back-off. But whenever there's a failure, also log and in the log message tell the user the command to run to make that permanode media's as ignored, so it's not tried again in the future.

**Comment by remram44** [comment]
I agree completely. On a related note, should 404 be considered a temporary error?

**Comment by mpl** [comment]
first draft
https://camlistore-review.googlesource.com/8588

**Comment by mpl** [comment]
@bradfitz what do you think of the story in https://camlistore-review.googlesource.com/8588?

@remram44 feel free to try out/review the CL. except if you have enough tweets that you need to use the zip archive (which I haven't yet considered). In which case, please just wait a bit more.

**Comment by mpl** [comment]
I have now added support for the zip case as well. But not sure how to test it yet, if that can even be done. I can probably tweak our testdata to include such a failure/recovery.

**Comment by mpl** [comment]
well, I had set up the https://twitter.com/camlitests for testing that CL, and I hadn't realized I could simply get the zip archive for it as well. retries seem to work as well in the "zip mode", so PTAL.
